python/pyspark/sql/functions.py

```diff
@@ -962,9 +993,9 @@ def add_months(start, months):
     """
     Returns the date that is `months` months after `start`
 
-    >>> df = spark.createDataFrame([('2015-04-08',)], ['d'])
-    >>> df.select(add_months(df.d, 1).alias('d')).collect()
-    [Row(d=datetime.date(2015, 5, 8))]
+    >>> df = spark.createDataFrame([('2015-04-08',)], ['dt'])
```

Comment:
In the original example, the column name `d` is reused for the new column. I think using a new column name helps differentiate the original date from the new date.